drivers/gpio/Kconfig.imx

```diff
@@ -5,6 +5,8 @@
 
 config GPIO_IMX
 	bool "IMX GPIO driver"
+	default y
 	depends on HAS_IMX_GPIO
```

Comment:
It would be nice to get rid of `HAS_IMX_GPIO` and `HAS_MCUX_IGPIO` as well. Could we add a dependency on a specific SoC here? And the same dependency negated for `HAS_MCUX_IGPIO`?

Reply:
I have a query to the NXP guys about what's going on here, since the two drivers are using the same compatible.  That implies we should only have one driver or need different compatibles.  Once we figure that out, will send a follow up PR to address what's going on.

Reply:
We need both drivers as `IMX_GPIO `is used by` imx` and `MCUX_IGPIO` by the mxrt families.

Reply:
Maybe we can have separate compatibles for the 2 drivers.

Reply:
> Maybe we can have separate compatibles for the 2 drivers.

We can, but the question if they are the same IP block, than can't we just use the IGPIO driver?

Reply:
`igpio` and `imx gpio` IP blocks are mostly similar, `igpio` has a few extra registers that is used for setting, clearing and toggling data bits. 
Also the Zephyr `igpio` driver is using MCUXpresso provided API's. We do not have an SDK for the imx SoC. I would recommend keeping these separate, to combine this we will have to move away from using the SDK and implement everything inside Zephyr.

Reply:
Ok, so probably best to introduce a new compatible/binding to distinguish them from each other.  Guessing rename the "i.mx" case and leaving "i.mx-rt" using `igpio`.